akka-http-core/src/main/scala/akka/http/impl/util/StreamUtils.scala

```diff
@@ -191,6 +191,56 @@ private[http] object StreamUtils {
     }
   }
 
+  /**
+   * INTERNAL API
+   *
+   * Returns a flow that is almost identity but delays propagation of cancellation from downstream to upstream.
+   */
+  def delayCancellation[T](cancelAfter: Duration): Flow[T, T, NotUsed] = Flow.fromGraph(new DelayCancellationStage(cancelAfter))
```

Comment:
This and below is just a revert of the previous removal of the `lingerTimeout` functionality.